lib/plugin/plugin.c

```diff
@@ -380,31 +384,72 @@ struct bt_component_class *bt_plugin_find_component_class(
 	return comp_cls;
 }
 
-/* Allocate dirent as recommended by READDIR(3), NOTES on readdir_r */
+static struct {
+	pthread_mutex_t lock;
+	struct bt_plugin_set *plugin_set;
+	bt_bool recurse;
+} append_all_from_dir_info = {
+	.lock = PTHREAD_MUTEX_INITIALIZER
+};
+
 static
-struct dirent *alloc_dirent(const char *path)
+int nftw_append_all_from_dir(const char *file, const struct stat *sb, int flag,
+		struct FTW *s)
 {
-	size_t len;
-	long name_max;
-	struct dirent *entry;
+	int ret = 0;
+	const char *name = file + s->base;
+
+	/* Check for recursion */
+	if (!append_all_from_dir_info.recurse && s->level > 1) {
+		goto end;
+	}
+
+	switch (flag) {
+	case FTW_F:
+		if (name[0] == '.') {
+			/* Skip hidden files */
+			BT_LOGV("Skipping hidden file: path=\"%s\"", file);
+			goto end;
+		}
+		struct bt_plugin_set *plugins_from_file =
```

Comment:
Move the variable declaration at the beginning of this `case` and enclose it within a scope to limit the variable's visibility:

```
[...]
case FOO:
{
    int bar;
}
[...]
```